src/pages/home/report/ReportView.js

```diff
@@ -4,40 +4,51 @@ import PropTypes from 'prop-types';
 import ReportActionsView from './ReportActionsView';
 import ReportActionCompose from './ReportActionCompose';
 import {addAction} from '../../../libs/actions/Report';
+import compose from '../../../libs/compose';
 import KeyboardSpacer from '../../../components/KeyboardSpacer';
 import styles from '../../../styles/styles';
 import SwipeableView from '../../../components/SwipeableView';
+import FullScreenLoadingIndicator from '../../../components/FullscreenLoading';
+import withDrawerState from '../../../components/withDrawerState';
+import withWindowDimensions from '../../../components/withWindowDimensions';
 
 const propTypes = {
-    // The ID of the report actions will be created for
+    /* The ID of the report the selected report */
     reportID: PropTypes.number.isRequired,
+
+    /* Is the view ready to be displayed */
+    isReady: PropTypes.bool.isRequired,
+
+    /* Is the report view covered by the drawer */
+    isDrawerOpen: PropTypes.bool.isRequired,
+
+    /* Is the window width narrow, like on a mobile device */
+    isSmallScreenWidth: PropTypes.bool.isRequired,
 };
 
-// This is a PureComponent so that it only re-renders when the reportID changes or when the report changes from
-// active to inactive (or vice versa). This should greatly reduce how often comments are re-rendered.
-class ReportView extends React.Component {
-    shouldComponentUpdate(prevProps) {
-        return this.props.reportID !== prevProps.reportID;
-    }
-
-    render() {
-        return (
-            <View style={[styles.chatContent]}>
-                <ReportActionsView
-                    reportID={this.props.reportID}
+function ReportView(props) {
+    const isComposeDisabled = props.isDrawerOpen && props.isSmallScreenWidth;
```

Comment:
Seems like perhaps this logic should be inside `ReportActionCompose` instead of the `ReportView`? e.g. we are passing `isDrawerOpen` to this component via the HOC and also passing `isDisabled` down. But can this all be contained in `ReportActionCompose` and reduce some logic+ extra props here?

Reply:
Yep, I'm cleaning up stuff and will move this